cmd/vicadmin/vicadm.go

```diff
@@ -426,12 +427,12 @@ func main() {
 	// If we're in an ESXi environment, then we need
 	// to extract the userid/password from UserPassword
 	if vchConfig.UserPassword != "" {
-		upw := strings.Split(vchConfig.UserPassword, ":")
-		if len(upw) == 2 {
-			vchConfig.Target.User = url.UserPassword(upw[0], upw[1])
-		} else {
-			vchConfig.Target.User = url.User(upw[0])
-		}
+		newurl, _ := url.Parse(fmt.Sprintf("%s://%s@%s%s",
+								vchConfig.Target.Scheme,
```

Comment:
Did your IDE push these lines so far over to the right? I find it a little uneasy on the eyes. :)